pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -69,10 +72,46 @@ func nodeSelectorMatchesThisNode(cl client.Client, eventObject runtime.Object) b
 	return matches(policyNodeSelector, node.ObjectMeta.Labels)
 }
 
+func initializeEnactment(client client.Client, policy *nmstatev1alpha1.NodeNetworkConfigurationPolicy) (*nmstatev1alpha1.NodeNetworkConfigurationEnactment, error) {
+
+	node := corev1.Node{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, &node)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("cannot get node %s", nodeName))
+	}
+
+	ownerRefList := []metav1.OwnerReference{
```

Comment:
Can we initialize the object in the Enactment constructor (under types.go) and handle the Get and Create here?

Reply:
no one else is going to create and enactment with owner refs here is good enough let's not add to much stuff at _types.go

Reply:
Fair enought. Would you mind splitting this function in here? So it is easier to test, one function creates an object with no side effects, the other calls API.

Reply:
At the end I have create a NewEnactment at _types.go it feels ok.